pmd-kotlin/src/test/resources/net/sourceforge/pmd/lang/kotlin/ast/testdata/Simple.txt

```diff
@@ -0,0 +1,270 @@
++- File
+   +- T-NL
+   +- T-NL
+   +- PackageHeader
+   |  +- T-PACKAGE
+   |  +- Identifier
+   |  |  +- SimpleIdentifier
+   |  |  |  +- T-Identifier
+   |  |  +- T-DOT
+   |  |  +- SimpleIdentifier
+   |  |  |  +- T-Identifier
+   |  |  +- T-DOT
+   |  |  +- SimpleIdentifier
+   |  |  |  +- T-Identifier
+   |  |  +- T-DOT
+   |  |  +- SimpleIdentifier
+   |  |  |  +- T-Identifier
+   |  |  +- T-DOT
+   |  |  +- SimpleIdentifier
+   |  |  |  +- T-Identifier
+   |  |  +- T-DOT
+   |  |  +- SimpleIdentifier
+   |  |  |  +- T-Identifier
+   |  |  +- T-DOT
+   |  |  +- SimpleIdentifier
+   |  |     +- T-Identifier
+   |  +- Semi
+   |     +- T-NL
+   |     +- T-NL
```

Comment:
That's how `package net.sourceforge.pmd.lang.kotlin.ast.testdata` is currently parsed. That's unfortunately very verbose and that's how the antlr grammar is written. That's why I want to mark this language as experimental, so that we can maybe simplify the AST somehow and don't worry about compatibility.

Reply:
Agreed. We use @joinTokenText in rules to address the AST verbosity, however, it is deprecated because of bad performance. Would be nice to have an alternative which keeps working with the improved AST.